Assets/Scripts/Utils/CursorBehaviour.cs

```diff
@@ -1,13 +1,37 @@
 ﻿using UnityEngine;
 
-public class CursorBehaviour : MonoBehaviour {
-	private void Start() {
-		Cursor.lockState = CursorLockMode.Locked;
-	}
-	
-	private void Update() {
-		
-		if(Input.GetKeyDown(KeyCode.LeftAlt))
-			Cursor.lockState = Cursor.lockState == CursorLockMode.None ? CursorLockMode.Locked : CursorLockMode.None;	
-	}
+/** CursorBehaviour Class
+ * This Class is used by Unity to display and Hide the mouse Cursor when the Left Alt button is pressed. 
+ * It Also ensures that the camera can not be moved when the player is using the mouse.
+ * Please note that, at the begining of the game, the cursor is Locked (hidden).
+ **/
+public class CursorBehaviour : MonoBehaviour
```

Comment:
Juste pour savoir, ne serait il pas une bonne idée de mettre un :

[RequireComponent(typeof(CameraController))]

pour s'assurer qu'un tel script n'est utilisé que pour bloquer la caméra et afficher le curseur ?

Reply:
J'y avais pas pensé, en fait le script est attaché au prefab du player, je sais pas si en mettant un RequireComponent ca va ajouter une camera ou prendre celle qui existe déjà. Je testerai dans les jours qui viennent et je posterai les résultats ici :)

Reply:
Très bien, alors c'est ok pour moi =)

Reply:
Après essaie, le [RequireComponent(typeof(CameraController))] va automatiquement ajouter le script si la cible ne le possède pas et ça fait la merde.
Il vaut donc mieux continuer de tester si not null.
On le saura pour la prochaine fois :)